core/domain/expression_parser.py

```diff
@@ -131,11 +158,23 @@ def tokenize(expression):
         else:
             token_list.append(Token(token_text))
 
-    if sum([len(token.text) for token in token_list]) != len(expression):
-        raise Exception(
-            'Invalid syntax: Unexpected token present in the expression.')
-
-    return token_list
+    # Adding '*' sign after identifiers, numbers and closing brackets if they
+    # are not followed by a valid operator.
+    final_token_list = []
+    for i in python_utils.RANGE(len(token_list)):
+        final_token_list.append(token_list[i])
+        if i != len(token_list) - 1:
+            # If a closing term is directly followed by a closing term, instead
+            # of an operator, we add the '*' operator.
```

Comment:
'instead of an operator'? Did you mean 'instead of no operator'?

Reply:
No, what I meant was that the term should be followed by an operator but we insert the * sign if it is followed by another term instead of an operator. I've updated the comment. Does that make things clearer, @aks681?

Reply:
Yup, looks better.